(docs): docs deployment for release branches

zhenmichael opened the pull request:
Note: requires changes from following PR https://github.com/microsoft/FluidFramework/pull/22252

Updated build-docs to trigger on release branches. 
Now includes a stage before deployment to check if triggering branch version matches a latest minor version of a major release.

zhenmichael committed:
include release branch in build-docs trigger, add check for latest release version

zhenmichael commented:
@tylerbutler I'm using the `$(RELEASE_VERSION)` in this PR. However, looking at current build-docs pipeline runs, it seems to list release version as 0.24, are there alternative variables with the correct version number?
![image](https://github.com/user-attachments/assets/19e19a08-ba6b-445f-b11d-4c27f535bf5c)

tylerbutler commented:
> @tylerbutler I'm using the `$(RELEASE_VERSION)` in this PR. However, looking at current build-docs pipeline runs, it seems to list release version as 0.24, are there alternative variables with the correct version number?

Hmmm, it's coming back to me now. This was a very early attempt at having multiple versions of the docs. I never finished the full thing, and we never actually published multiple versions or had multiple versioned pipelines running afaik. If you want to repurpose those variables and update them, that should be fine.

zhenmichael committed:
giMerge branch 'main' into build-docs-deployment-update

zhenmichael committed:
Merge branch 'main' into build-docs-deployment-update

zhenmichael committed:
Merge branch 'main' into build-docs-deployment-update

zhenmichael committed:
run generate buildVersion in build-docs

zhenmichael committed:
Merge branch 'main' into build-docs-deployment-update

zhenmichael committed:
include package name in flub command

tylerbutler commented on tools/pipelines/build-docs.yml:
```diff
@@ -301,10 +301,30 @@ stages:
+            workingDirectory: ${{ parameters.buildDirectory }}
+            script: |
+
+              # Generate the build version. Sets the environment variables version, codeVersion, and isLatest.
+              flub generate buildVersion
```
Double-check that the include-vars.yml workflow is included in this pipeline. Those vars are all required for generate:buildVersion to work correctly.

zhenmichael committed:
add install build-tools step and separate generate buildVersion step

zhenmichael committed:
update flub command to set shouldDeploy variable

zhenmichael commented:
Added include-install-build-tools.yml template before running generate buildVersion and it seems to run as expected in the test pipeline: https://dev.azure.com/fluidframework/internal/_build/results?buildId=296943&view=logs&j=332c3c5d-33c7-5b70-ec97-9939bb462e3a&t=7863b927-2c80-5032-566b-28c68c74517f

I also adjusted the latestVersions command to set a `shouldDeploy` variable instead of exiting with error. Seems to be working as expected from test pipeline run: https://dev.azure.com/fluidframework/internal/_build/results?buildId=296950&view=logs&j=332c3c5d-33c7-5b70-ec97-9939bb462e3a&t=19081285-5379-557e-c1b4-28ee287685b8

tylerbutler commented on build-tools/packages/build-cli/src/commands/check/latestVersions.ts:
```diff
@@ -62,21 +62,21 @@ export default class LatestVersionsCommand extends BaseCommand<typeof LatestVers
 					`skipping deployment stage. input version ${versionInput.version} does not match the latest version ${v.version}`,
-					{ exit: 1 },
 				);
+				this.log(`##vso[task.setvariable variable=shouldDeploy;isOutput=true]false`);
+				return;
```
I'm conflicted about this approach. Maybe formatting the messages as warnings using the info here would be enough? https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#formatting-commands

otherwise where how can we tell if the step succeeded?

tylerbutler reviewed (approved):
I'm fine with this, but we may need to revisit the logging/reporting parts if it confuses folks.

zhenmichael committed:
log warning update
